js/components/matrix.js

```diff
@@ -0,0 +1,343 @@
+          {
+            label: T('widgets.form.matrix.defaultOptions.row1')
+          },
+          {
+            label: T('widgets.form.matrix.defaultOptions.row2')
+          }
+        ];
+      }
+    },
+    columnOptions: {
+      type: Array,
+      default: function() {
+        return [
+          {
+            label: T('widgets.form.matrix.defaultOptions.column1')
+          },
+          {
+            label: T('widgets.form.matrix.defaultOptions.column2')
+          }
+        ];
+      }
+    }
+  },
+  watch: {
+    value: function(val) {
+      if (this.checkValue(val) === 'clear') {
+        this.clearValue();
+      } else {
+        this.setValue();
+      }
+    }
+  },
+  methods: {
+    /**
+     * Returns unique ID to the row with combination of form widget instance, name, rowIndex and columnIndex.
+     * @param {Number} rowIndex - a row index
+     * @param {Number} columnIndex - a column index
+     * @param {String} type - an element type
+     * @returns {String} an ID unique to the row
+     */
+    getOptionId: function(rowIndex, columnIndex, type) {
+      return _.kebabCase(this.$parent.id + '-' + this.name + '-' + rowIndex + '-' + columnIndex + '-' + type);
+    },
+
+    /**
+     * Returns unique name to the row with combination of name and rowIndex.
+     * @param {Number} rowIndex - a row index
+     * @returns {String} a name unique to the row
+     */
+    getOptionName: function(rowIndex) {
+      return _.kebabCase(this.name + '-' + rowIndex);
+    },
+
+    /**
+     * Click handler for each radio button in matrix
+     * @param {Object} row - a row object which includes id and label
+     * @param {Object} column - a column object which includes id and label
+     * @param {Number} rowIndex - a row index
+     * @param {Number} colIndex - a column index
+     * @returns {undefined}
+     */
+    clickHandler: function(row, column, rowIndex, colIndex) {
+      if (rowIndex >= 0 && colIndex >= 0) {
+        var el = this.getOptionId(rowIndex, colIndex, 'input');
+
+        $('#' + el).prop('checked', true);
+      }
+
+      if (row.id && this.value[row.id]) {
+        delete this.value[row.id];
+      }
+
+      if (this.value[row.label]) {
+        delete this.value[row.label];
+      }
+
+      this.value[row.id || row.label] = column ? column.id || column.label : undefined;
+
+      this.$emit('_input', this.name, this.value);
+    },
+
+    /**
+     * Focus handler for each radio button in matrix
+     * @param {Number} rowIndex - a row index
+     * @param {Number} colIndex - a column index
+     * @returns {undefined}
+     */
+    focusHandler: function(rowIndex, colIndex) {
+      var $vm = this;
+
+      if (rowIndex >= 0 && colIndex >= 0) {
+        var row = this.rowOptions[rowIndex];
+        var col = this.columnOptions[colIndex];
+
+        $('#' + $vm.getOptionId(rowIndex, colIndex, 'input')).focus();
+
+        $('#' + $vm.getOptionId(rowIndex, colIndex, 'span')).focus();
```

Comment:
It doesn't seem to make sense to have two instructions running `.focus()` consecutively. Surely they can't both be necessary.